### ISSUE TITLE
Layer's related `aria-haspopup` misplaced/misused

### ISSUE THREAD

**Issue by Malvoz** [Leaflet/Leaflet#7192]
Currently, `aria-haspopup` is improperly set on the layer list's and control's container `div` element:

https://github.com/Leaflet/Leaflet/blob/983092a4975e5a4ab74e8665b38e31a17f38ddb5/src/control/Control.Layers.js#L180

The [`aria-haspopup`](https://www.w3.org/TR/wai-aria-1.1/#aria-haspopup) description suggests that the attribute should be set on the element that triggers additional content (which is `<a class="leaflet-control-layers-toggle">` in this case).

> Indicates the availability and type of interactive popup element, such as menu or dialog, that can be triggered by an [element](https://www.w3.org/TR/wai-aria-1.1/#dfn-element).

It might be a good idea to check out the different patterns in the W3C ARIA Practices documentation to establish what sort of pattern would best fit the layer control (for example the [disclosure](https://www.w3.org/TR/wai-aria-practices/#disclosure) widget).

**Comment by johnd0e** [comment]
> Currently, `aria-haspopup` is improperly set on the layer list's and control's container div element:

See https://github.com/Leaflet/Leaflet/commit/d18f6249b2bf1db4e0b4be42124fb94303aa99c0.
It's sole function is proper functioning on Windows phones, and it seems ok.

**Comment by Malvoz** [comment]
@johnd0e Interesting, I found the [explainer](https://docs.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/dev-guides/jj152135(v=vs.85)) from Microsoft (and related mailing list conversations [W3C](https://lists.w3.org/Archives/Public/wai-xtech/2013Dec/0008.html), [WebAIM](https://webaim.org/discussion/mail_thread?thread=6255)).

The issues withstanding is that the triggering element does not have `aria-haspopup`, which per definition "indicates the availability" of popup content.

Also, once the layer toggle is pressed, there's no announcement that new content has been made available to the user. Looking at the [disclosure patterns examples](https://www.w3.org/TR/wai-aria-practices-1.2/#disclosure) it seems `aria-expanded` and `aria-controls` should be used to convey that.

**Comment by Malvoz** [comment]
Even if `aria-haspopup` was properly conveyed to users of assistive technology by moving it to the `<a>` control, it would be incorrect to do so because `aria-haspopup` has requirements of the type of content it controls:

[`aria-haspopup`](https://www.w3.org/TR/wai-aria-1.2/#aria-haspopup):
> Authors _**MUST**_ ensure that the role of the element that serves as the container for the popup content is [`menu`](https://www.w3.org/TR/wai-aria-1.2/#menu), [`listbox`](https://www.w3.org/TR/wai-aria-1.2/#listbox), [`tree`](https://www.w3.org/TR/wai-aria-1.2/#tree), [`grid`](https://www.w3.org/TR/wai-aria-1.2/#grid), or [`dialog`](https://www.w3.org/TR/wai-aria-1.2/#dialog), and that the value of `aria-haspopup` matches the role of the popup container.

None of which I believe are appropriate for the layer controls list.

I'm closing this, superseded by https://github.com/Leaflet/Leaflet/issues/7526.
